File: stats_series_01_week_12.R
# Snippet 1: Introduction to Time Series and Components
library(stats)
decomposition <- decompose(ts(time_series, frequency=12))

# Snippet 2: Stationarity and Differencing
library(tseries)
diff_series <- diff(time_series)
adf_test <- adf.test(diff_series)

# Snippet 3: ARIMA Models
fit <- arima(time_series, order=c(1, 1, 1))

# Snippet 4: Advanced Time Series Methods and Forecasting
library(forecast)
holt_winters_model <- HoltWinters(ts(time_series, frequency=12))
fit_hw <- forecast(holt_winters_model)

File: stats_series_01_week_12.py
# Snippet 1: Introduction to Time Series and Components
import statsmodels.api as sm

decomposition = sm.tsa.seasonal_decompose(time_series, model='additive')

# Snippet 2: Stationarity and Differencing
from statsmodels.tsa.stattools import adfuller

diff_series = time_series.diff().dropna()
adf_test = adfuller(diff_series)

# Snippet 3: ARIMA Models
from statsmodels.tsa.arima_model import ARIMA

model = ARIMA(time_series, order=(1, 1, 1))
fit = model.fit()

# Snippet 4: Advanced Time Series Methods and Forecasting
from statsmodels.tsa.holtwinters import ExponentialSmoothing

holt_winters_model = ExponentialSmoothing(time_series, seasonal='additive', seasonal_periods=12)
fit_hw = holt_winters_model.fit()
